### ISSUE TITLE
Open Form Issues

### ISSUE THREAD

**Issue by MartinBernard** [hres/pmp#62]
A) Peter B to provide following [ see #42 ]

(1) XML Prolog
(2) Document Declaration
(3) Document Definition
(4) Codes

1) Add Expand/Collapse All Sections Function

2) Table Issues for adding Columns (ref #25)

3) Char Limit -  fields are locked at 1500 chars. it will need to be revised.

4) We still need the ability to add on demand tables and images to each section. For example, Part II Toxicology section, often we'll get tables there.

5) Ability to add columns to tables - still an issue. see item #67 sub item 4 for more info.

6) Ability to add complete section for Part III so we can capture instance where multiple pm part III are found. [e.g.: DOVOBET® Gel & DOVOBET® Gel with Applicator]

7) File naming convention will when saving XML will need to change. Currently, it saves all files as "productmonograph.zip"

8) Timeout. Can you please remove the timeout session on the application. I was under the impression this was already done but apparently under iis-rep.hres.ca, testers are still experiencing this problem.

9)

**Comment by MartinBernard** [comment]
@dbeauvai  / @OttawaOn -- pls note action items

@peterbomberg - any thoughts on item 7 above?

**Comment by peterbomberg** [comment]
YEs #7 should be updated to be the document id followed by “.xml”, however at this time I don't think the id element in the document information section of the SPL file is being set by the application, which means this is rather hard to do.

**Comment by MartinBernard** [comment]
@dbeauvai / @OttawaOn - for now, please amend to <brandname.zip> format until document id is generate

**Comment by MartinBernard** [comment]
@dbeauvai - please get Ching to focus on the item 4 to 7 as listed above before addressing any web standards #39 items . We need to correct those issues before we formally reach out to industry for the proof of concept testing from external participants. We intend to reach out to participants early February so time is of the essence. Web Standards fixes can wait.

**Comment by MartinBernard** [comment]
> 5) Ability to add columns to tables - still an issue. see item #67 sub item 4 for more info.

Changes is required for all tables throughout the form. Some examples, but not limited to, are under Part I Clinical Trials and Drug Interactions

> 6) Ability to add complete section for Part III so we can capture instance where multiple pm part III are found. [e.g.: DOVOBET® Gel & DOVOBET® Gel with Applicator]

You will need to create an ADD / REMOVE button for the complete PART III. 

> 7) File naming convention when saving XML will need to change. Currently, it saves all files as "productmonograph.zip". We should take the user entry from the brand name field on the first page to generate the file name. For example, if Advil is inserted int he Brand Name field then saving its draft will generate the following file "advil.zip".

**Comment by dbeauvai** [comment]
@OttawaOn  - Please work on #7, #6 and #5 as discussed.  If you have any questions please contact me.

**Comment by dbeauvai** [comment]
@MartinBernard  - for #7.  What is the rule if they have not entered the brand name yet?  Shall we put as default productmonograph or else?

**Comment by MartinBernard** [comment]
Pls note that additional work issues will be added to this list.

As for item 7 herein, up to you I'm indifferent how we name it, it cannot be however productmonograph.xml

**Comment by MartinBernard** [comment]
@dbeauvai - please add item 8 above to Ching workload. As mentioned, more are forthcoming

**Comment by peterbomberg** [comment]
7) File naming convention will when saving XML will need to change. Currently, it saves all files as "productmonograph.zip"

The file name must be unique as a submission can contain more than one product. So the brand name .xml would work for now however the long term aim is to use the document id@root value.xml

**Comment by MartinBernard** [comment]
item b) added.

**Comment by OttawaOn** [comment]
Hi!
Saved-file name is already changed into Brandname.xml e.g myBrandname.xml and compressed as Brandname.zip. e.g: myBrandname.zip
if no brandname, we use draftPMForm.xml and draftPMForm.zip as template filename.

**Comment by MartinBernard** [comment]
@dbeauvai ---- not sure @OttawaOn is clear on the requirements we discussed. Please clarify with her. We will discuss on Wednesday.

**Comment by dbeauvai** [comment]
Status:

Ching has worked very hard to make the table compliant to WET 4 and to Append Rows and Columns.  She did it for one table and we will submit it to the Web Standard group to make sure we are compliant then she will do all the other one.  She is presently working on the function to save the result of the table in the session and in the xml file.  

The function to save was working but it was affected by the change she did in the table.  Don't be surprise if the save draft is not working as it was.  She is fixing it.

**Comment by dbeauvai** [comment]
We did not make as much progress as we wanted because the task of adding columns and rows was complex.  The developer had to redo what was done before in order to be compliant to WRI. 
We had issues saving to github and getting the proper result on iis.  Lost a lot of time debugging but it is now resolved.  PART II and PART III are almost all bilingual.  The code will be put on github/iis tomorrow when I am in Kanata. 

7) File naming convention will when saving XML will need to change. Currently, it saves all files as "productmonograph.zip" Ready – you can test on iis-rep.hres.ca

6) Ability to add complete section for Part III so we can capture instance where multiple pm part III are found. [e.g.: DOVOBET® Gel & DOVOBET® Gel with Applicator]  In progress

5) Ability to add columns to tables - still an issue. see item #67 sub item 4 for more info. - In progress 
This task is very complex but it is looking good now.

4) We still need the ability to add on demand tables and images to each section. For example, Part II Toxicology section, often we'll get tables there. - not done yet 

8) Timeout. Can you please remove the timeout session on the application. I was under the impression this was already done but apparently under iis-rep.hres.ca, testers are still experiencing this problem.    - not done yet

First Page Changes #76

1) Ability to input multiple active ingredients and corresponding strengths for one tablets. For example, see PM ATRIPLA. We need to be able to capture these. In other words, we need an ADD/REMOVE button wrapped around the first table to be able to capture multiple ingredients.   

Billingualism - Making application bilingual    In progress – 35%

Part II and Part III almost completed 

WRI - Fixing code for web renewal initiative    In progress – in parallel

**Comment by OttawaOn** [comment]
I finished the program part of "Save Draft File" with file name, now it
works. also finish functions on Add Row and Remove Row, and save the data
of new row into the XML file and Session.

**Comment by MartinBernard** [comment]
@peterbomberg ----  can you please add the item we talked about this afternoon and assign to Diane. That is the requirements to generate and render a recognized SPL file after its transformation.  Either document here as item 10 or open another ticket.

**Comment by MartinBernard** [comment]
> 8) Timeout. Can you please remove the timeout session on the application. I was under the impression this was already done but apparently under iis-rep.hres.ca, testers are still experiencing this problem. - not done yet

pls set at 5hrs timeout period

**Comment by dbeauvai** [comment]
@MartinBernard  - The IIS server idle-timeout was changed to 300 minutes (5hrs). - Timeout issue is now done

**Comment by MartinBernard** [comment]
(10)  http://iis-rep.hres.ca/ does not work with chrome #81 works fine on IE but the load template button is not functioning on Chrome (Windows 10)

**Comment by MartinBernard** [comment]
(11) Make form Bilingual (reference #70 for background) . We will track here from now on.

**Comment by MartinBernard** [comment]
Ref 76 - we will track here moving forward

(12) Ability to input multiple active ingredients and corresponding strengths for one tablets. For example, see PM ATRIPLA. We need to be able to capture these. In other words, we need an ADD/REMOVE button wrapped around the first table to be able to capture multiple ingredents.

PM00030932.pdf

(13) We also need to ability to capture joint venture agreements where two companies own the product etc. so possibly and ADD/REMOVE button for sponsor name and sponsor address.

**Comment by MartinBernard** [comment]
@dbeauvai - we will need to discuss this urgently |(as in tomorrow). As I\m going through the form again, I catch myself going thru issues that were reported and addressed with Mike. The look and feel, format is also not optimal. We'll need to have an honest and frank discussion about the progress. I've already invested hours?month reporting bug/fixes, etc., I can't go through that exercise again. We'll need to discuss a better approach b/c it's not working.

**Comment by MartinBernard** [comment]
Will be tracked and reported using spreadsheet.

**Comment by OttawaOn** [comment]
Yes, I get it on file name, later we will create filename as document
id@root value.xml, and open the client folder...
Now The cover page was finished as bilingual. I am working on Part I...
Ching
